src/Compilers/CSharp/Portable/Binder/PatternVariableFinder.cs

```diff
@@ -13,51 +13,42 @@ namespace Microsoft.CodeAnalysis.CSharp
     class PatternVariableFinder : CSharpSyntaxWalker
     {
         ArrayBuilder<DeclarationPatternSyntax> declarationPatterns;
-        ArrayBuilder<ExpressionSyntax> expressionsToVisit = ArrayBuilder<ExpressionSyntax>.GetInstance();
+        ArrayBuilder<CSharpSyntaxNode> nodesToVisit = ArrayBuilder<CSharpSyntaxNode>.GetInstance();
```

Comment:
This is a preexisting condition common to pattern matching implementation (CC @gafter ). I will open an issue to track this. For stabilization, I'd rather keep the churn to a minimum and follow the style used in this class.